floors/enforce_test.go

```diff
@@ -0,0 +1,1158 @@
+				},
+				{
+					Seat:     "appnexus",
+					Currency: "USD",
+					Bids:     []*entities.PbsOrtbBid{{Bid: &openrtb2.Bid{ID: "some-bid-11", Price: 0.5, DealID: "deal_Id_3", ImpID: "some-impression-id-1"}, BidFloors: &openrtb_ext.ExtBidPrebidFloors{FloorCurrency: "USD", FloorValue: 20.01}}},
+				},
+			},
+			expErrs: []error{},
+		},
+		{
+			name: "Should enforce floors when imp.bidfloor provided and enforcepbs not provided",
+			args: args{
+				bidRequestWrapper: &openrtb_ext.RequestWrapper{
+					BidRequest: &openrtb2.BidRequest{
+						ID: "some-request-id",
+						Imp: []openrtb2.Imp{{
+							ID: "some-impression-id-1", Banner: &openrtb2.Banner{Format: []openrtb2.Format{{W: 300, H: 250}}}, BidFloor: 5.01, BidFloorCur: "USD",
+						}},
+						Ext: json.RawMessage(`{"prebid":{"floors":{"floormin":1}}}`),
+					},
+				},
+				seatBids: map[openrtb_ext.BidderName]*entities.PbsOrtbSeatBid{
+					"pubmatic": {
+						Bids: []*entities.PbsOrtbBid{
+							{Bid: &openrtb2.Bid{ID: "some-bid-1", Price: 1.2, DealID: "deal_Id_1", ImpID: "some-impression-id-1"}},
+						},
+						Seat:     "pubmatic",
+						Currency: "USD",
+					},
+					"appnexus": {
+						Bids: []*entities.PbsOrtbBid{
+							{Bid: &openrtb2.Bid{ID: "some-bid-11", Price: 4.5, ImpID: "some-impression-id-1"}},
+						},
+						Seat:     "appnexus",
+						Currency: "USD",
+					},
+				},
+				conversions:    convert{},
+				priceFloorsCfg: config.AccountPriceFloors{Enabled: true, EnforceFloorsRate: 100, EnforceDealFloors: false},
+			},
+			expEligibleBids: map[openrtb_ext.BidderName]*entities.PbsOrtbSeatBid{
+				"pubmatic": {
+					Bids: []*entities.PbsOrtbBid{
+						{Bid: &openrtb2.Bid{ID: "some-bid-1", Price: 1.2, DealID: "deal_Id_1", ImpID: "some-impression-id-1"}, BidFloors: &openrtb_ext.ExtBidPrebidFloors{FloorValue: 5.01, FloorCurrency: "USD"}},
+					},
+					Seat:     "pubmatic",
+					Currency: "USD",
+				},
+				"appnexus": {
+					Bids:     []*entities.PbsOrtbBid{},
+					Seat:     "appnexus",
+					Currency: "USD",
+				},
+			},
+			expRejectedBids: []*entities.PbsOrtbSeatBid{
+				{
+					Seat:     "appnexus",
+					Currency: "USD",
+					Bids:     []*entities.PbsOrtbBid{{Bid: &openrtb2.Bid{ID: "some-bid-11", Price: 4.5, ImpID: "some-impression-id-1"}, BidFloors: &openrtb_ext.ExtBidPrebidFloors{FloorValue: 5.01, FloorCurrency: "USD"}}},
+				},
+			},
+			expErrs: []error{},
+		},
+	}
+	for _, tt := range tests {
+		actEligibleBids, actErrs, actRejecteBids := Enforce(tt.args.bidRequestWrapper, tt.args.seatBids, config.Account{PriceFloors: tt.args.priceFloorsCfg}, tt.args.conversions)
+		assert.Equal(t, tt.expErrs, actErrs, tt.name)
+		assert.Equal(t, len(tt.expEligibleBids), len(actEligibleBids), "Number of eligible bids mismatch for test "+tt.name)
+		assert.Equal(t, len(tt.expRejectedBids), len(actRejecteBids), "Number of rejected bids mismatch for test "+tt.name)
```

Comment:
I recommend the following so you can still verify the bid data structures contain the right information:
```
assert.ElementsMatch(t, tt.expRejectedBids, actEligibleBids)
if reflect.DeepEqual(tt.expEligibleBids, actEligibleBids) {
	assert.Fail(t, "eligible bids don't match")
}
```

For a map, the `testify/assert` package doesn't provide a function that verifies map equality so `reflect` should be ok here.

Reply:
Done